server/src/main/java/org/elasticsearch/action/admin/indices/alias/get/TransportGetAliasesAction.java

```diff
@@ -63,6 +67,17 @@ protected GetAliasesResponse newResponse() {
     protected void masterOperation(GetAliasesRequest request, ClusterState state, ActionListener<GetAliasesResponse> listener) {
         String[] concreteIndices = indexNameExpressionResolver.concreteIndexNames(state, request);
         ImmutableOpenMap<String, List<AliasMetaData>> result = state.metaData().findAliases(request.aliases(), concreteIndices);
-        listener.onResponse(new GetAliasesResponse(result));
+
+        // in case all aliases are requested then it is desired to return the concrete index with no aliases (#25114):
+        boolean aliasesProvided = Strings.isAllOrWildcard(request.getOriginalAliases());
+        ImmutableOpenMap.Builder<String, List<AliasMetaData>> mapBuilder = ImmutableOpenMap.builder(result);
+        Iterable<String> intersection = HppcMaps.intersection(ObjectHashSet.from(concreteIndices), state.metaData().indices().keys());
```

Comment:
I am not sure why we need the intersection here. We have just resolved the indices to concrete indices based on the same cluster state, so it should all be existing indices?